tests/test_query.py

```diff
@@ -114,6 +117,72 @@ def mock_httpx_post(**kwargs):
     assert response.status_code == 401
 
 
+def test_app_with_unset_allowed_origins(test_app, monkeypatch):
+    """Tests that when the environment variable for allowed origins has not been set, a warning is raised and the app uses a default value."""
+    monkeypatch.delenv(util.ALLOWED_ORIGINS.name, raising=False)
+    # set random username and password to avoid RuntimeError from other startup check
+    monkeypatch.setenv(util.GRAPH_USERNAME.name, "DBUSER")
+    monkeypatch.setenv(util.GRAPH_PASSWORD.name, "DBPASSWORD")
+
+    with pytest.warns(
+        UserWarning,
+        match=f"API was launched without providing any values for the {util.ALLOWED_ORIGINS.name} environment variable",
+    ):
+        with test_app:
+            pass
+
+    assert util.parse_origins_as_list(
```

Comment:
Why is this part of the test? Aren't you implicitly testing the monkeypatch function here? For the `parse_origins_as_list` portion the next test is a better fit.

Reply:
The purpose of this test is to check, when the `NB_API_ALLOWED_ORIGINS` environment variable is undefined, that the API: 
- a) emits a warning (this warning is also emitted when `NB_API_ALLOWED_ORIGINS` is explicitly set to `""` by the user)
- b) internally assigns a default value of `""` for the variable, meaning the CORS middleware ends up with `[""]` as the list of allowed origins

The assertion above is to explicitly check b), that is, that the final list of origins isn't something weird just because the environment variable wasn't set.